packages/sns/src/swap.canister.ts

```diff
@@ -36,4 +38,16 @@ export class SwapCanister extends Canister<SnsSwapCanister> {
   ): Promise<void> => {
     await this.caller({ certified: true }).refresh_buyer_tokens(params);
   };
+
+  /**
+   * Get user commitment
+   */
+  getUserCommitment = async (
+    params: GetBuyerStateRequest & QueryParams
```

Comment:
- is it worth creating a type `GetUserCommitmentParams` in a new `swap.params.ts` - as for example `governance.params`?

- what's on the contrary sure is that either that new types `GetUserCommitmentParams` or `GetBuyerStateRequest` should be exposed in `index.ts` otherwise you won't be able to use it in nns-dapp

Reply:
Well, I thought that if the idea is to always keep the same types from sns-js, we should do the same for the request types, no?

Reply:
Yes but it's the type + `{certified: boolean}` anyway.

I just meant for example:

```
export type  GetUserCommitmentParams = GetBuyerStateRequest & QueryParams
```

but I'm fine as it is too, not strong opinion. On the contrary, a type definitely need to be exposed in `index.ts`

Reply:
let's just expose `GetBuyerStateRequest` and all good

Reply:
done!

Reply:
thx